boutdata/collect.py

```diff
@@ -107,18 +107,19 @@ def _convert_to_nice_slice(r, N, name="range"):
     elif len(r) == 2:
         r2 = list(r)
         if r2[0] < 0:
-            r2[0] += N
+            r2[0] = r2[0] + N
```

Comment:
Is this change just to make it more verbose what's going on?

Reply:
It's part of 178fb4f, which is a bug fix (before that chunks of 0 or Nan (not sure which) were being written on processors after the upper target). I wrote it long enough ago that I can't remember now which was the critical part. I think at least one of the places where it replaced `+=` with an assignment was important, and I changed the rest of them at the same time for consistency, and to avoid potential similar surprises.

Reply:
I'm pretty sure the bug wasn't here before this PR, it actually turned up during writing this where at one point I had this
```
            # Check lower y boundary
            if pe_yind == 0:
                # Keeping inner boundary
                if ystop <= 0:
                    inrange = False
                if ystart < 0:
                    ystart = 0
            else:
                if ystop < myg-1:
                    inrange = False
                if ystart < myg:
                    ystart = myg
            # and lower y boundary at upper target
            if yproc_upper_target is not None and pe_yind - 1 == yproc_upper_target:
                ystart -= myg
```
where `ystart = myg` didn't make a copy, so `ystart -= myg` could actually end up modifying `myg`, which caused the bug. Then I decided that it was safer not to use the modify-in-place operators anywhere to prevent any chance of this sort of behaviour.

Reply:
Wow. How did I manage to overlook such a big feature.
I'll add a links if anyone is interested
https://stackoverflow.com/questions/41446833/what-is-the-difference-between-i-i-1-and-i-1-in-a-for-loop